app/components/sidebar/index.ts

```diff
@@ -74,12 +74,50 @@ export default class SidebarComponent extends Component {
     this.renderer.toggleLeftSidebar(true);
   };
 
+  handlePanmoveInner = ({ gesture }: GestureEvent) => {
+    if (
+      !gesture.touchMoveX ||
+      (this.settings.isRightSidebar() && gesture.touchMoveX < 0) ||
+      (!this.settings.isRightSidebar() && gesture.touchMoveX > 0) ||
+      (gesture.touchMoveX && Math.abs(gesture.touchMoveX) > this.maxWidth)
+    ) {
+      return;
+    }
+
+    this.renderer.dragLeftSidebar(this.maxWidth - Math.abs(gesture.touchMoveX));
+
+    gesture.on('panend', () => {
+      this.renderer.toggleLeftSidebar(false);
+    });
+  };
+
+  handlePanmoveOuter = ({ gesture }: GestureEvent) => {
+    if (
+      !gesture.touchMoveX ||
+      (this.settings.isRightSidebar() && gesture.touchMoveX > 0) ||
+      (!this.settings.isRightSidebar() && gesture.touchMoveX < 0) ||
+      (gesture.touchMoveX && Math.abs(gesture.touchMoveX) > this.maxWidth)
+    ) {
+      return;
+    }
+
+    this.renderer.dragLeftSidebar(Math.abs(gesture.touchMoveX));
+
+    gesture.on('panend', () => {
+      this.renderer.toggleLeftSidebar(false);
+    });
+  };
+
   gestures: Record<string, Gesture[]> = {
     inner: [
       {
         type: 'swipeleft',
         onGesture: this.handleSwipeInner,
       },
```

Comment:
```suggestion
    inner: [
```
Können die alten Gesten nicht weg dann?

Reply:
Ne, tatsächlich nicht. Die handlen immernoch die allgemeine Swipe-Bewegung und damit das Öffnen/Schließen der Sidebar.

Bei den Swipe Gesten scheint es allerdings nur ein `start` und `end` zu geben unter der Haube.
Das Panmove scheint öfter zu "feuern" und man kann damit die Sidebar an den Finger kleben (quasi).

Ich müsste es mir sonst nochmal genauer angucken und checken, ob man auch drauf verzichten kann, denke aber eher nicht.
Die Doku von Tinygesture ist leider sehr mau.

Reply:
@spuxx1701 Gerade nochmal rumgespielt, es geht auch ohne die swipe gestures, allerdings verhält es sich gefühlt (im Browser getestet) besser mit den swipe gestures.

Zusätzlich wird teilweise bei zu viel hin- und her gedragge ohne die swipe gestures irgendwelche Endpunkte aufgerufen. Als ob irgendwas zig mal geklickt wird.
Hab mal Screens gemacht:

![Bildschirmfoto 2024-07-17 um 16 48 13](https://github.com/user-attachments/assets/0dde3737-238a-4ce6-a6f0-30ae2360918e)

![Bildschirmfoto 2024-07-17 um 16 48 42](https://github.com/user-attachments/assets/2e55bf54-2e01-4911-98f1-8cab13c9e3c4)

Kann die Version ohne Swipes auch nochmal pushen, wenn du willst. Würd sonst erstmal dafür plädieren die Swipes drin zu lassen 👀

Reply:
Oder du probierst mal bei dir lokal - so wäre die Version ohne Swipes:

```typescript
// app/components/sidebar/index.ts

  handlePanmoveInner = ({ gesture }: GestureEvent) => {
    if (
      !gesture.touchMoveX ||
      (this.settings.isRightSidebar() && gesture.touchMoveX < 0) ||
      (!this.settings.isRightSidebar() && gesture.touchMoveX > 0) ||
      (gesture.touchMoveX && Math.abs(gesture.touchMoveX) > this.maxWidth)
    ) {
      return;
    }

    this.renderer.dragLeftSidebar(this.maxWidth - Math.abs(gesture.touchMoveX));

    gesture.on('panend', () => {
      if (
        gesture.touchMoveX &&
        Math.abs(gesture.touchMoveX) > this.maxWidth / 2
      ) {
        this.renderer.toggleLeftSidebar(false);
      } else {
        this.renderer.toggleLeftSidebar(true);
      }
    });
  };

  handlePanmoveOuter = ({ gesture }: GestureEvent) => {
    if (
      !gesture.touchMoveX ||
      (this.settings.isRightSidebar() && gesture.touchMoveX > 0) ||
      (!this.settings.isRightSidebar() && gesture.touchMoveX < 0) ||
      (gesture.touchMoveX && Math.abs(gesture.touchMoveX) > this.maxWidth)
    ) {
      return;
    }

    this.renderer.dragLeftSidebar(Math.abs(gesture.touchMoveX));

    gesture.on('panend', () => {
      if (
        gesture.touchMoveX &&
        Math.abs(gesture.touchMoveX) > this.maxWidth / 2
      ) {
        this.renderer.toggleLeftSidebar(true);
      } else {
        this.renderer.toggleLeftSidebar(false);
      }
    });
  };

  gestures: Record<string, Gesture[]> = {
    inner: [
      {
        type: 'panmove',
        onGesture: this.handlePanmoveInner,
      },
    ],
    outer: [
      {
        type: 'panmove',
        onGesture: this.handlePanmoveOuter,
      },
    ],
  };
```

Reply:
Ich gucks mir mal an wenn ich Zeit hab, will aber gar nicht zu viel Zeit in die gestures stecken. Denke ich bringe den Stand dann erstmal auf staging und teste das mal bissi mit dem handy

Reply:
Passt für mich, würds mir auch gern mal aufm Handy anschauen.